persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/DiscreteCopyNumberSparkRepository.java

```diff
@@ -0,0 +1,167 @@
+
+    @Autowired
+    private SparkSession spark;
+
+    @Autowired
+    private ParquetLoader parquetLoader;
+    
+    @Autowired
+    private GeneRepository geneRepository;
+    
+    @Autowired
+    @Qualifier("molecularProfileMyBatisRepository")
+    private MolecularProfileRepository molecularProfileRepository;
+    
+    @Override
+    public List<DiscreteCopyNumberData> getDiscreteCopyNumbersInMolecularProfileBySampleListId(String molecularProfileId, String sampleListId, List<Integer> entrezGeneIds, List<Integer> alterationTypes, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaDiscreteCopyNumbersInMolecularProfileBySampleListId(String molecularProfileId, String sampleListId, List<Integer> entrezGeneIds, List<Integer> alterationTypes) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<DiscreteCopyNumberData> fetchDiscreteCopyNumbersInMolecularProfile(String molecularProfileId, List<String> sampleIds, List<Integer> entrezGeneIds, List<Integer> alterationTypes, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<DiscreteCopyNumberData> getDiscreteCopyNumbersInMultipleMolecularProfiles(List<String> molecularProfileIds, List<String> sampleIds, List<Integer> entrezGeneIds, List<Integer> alterationTypes, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta fetchMetaDiscreteCopyNumbersInMolecularProfile(String molecularProfileId, List<String> sampleIds, List<Integer> entrezGeneIds, List<Integer> alterationTypes) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<CopyNumberCountByGene> getSampleCountByGeneAndAlterationAndSampleIds(String molecularProfileId, List<String> sampleIds, List<Integer> entrezGeneIds, List<Integer> alterations) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<CopyNumberCountByGene> getSampleCountInMultipleMolecularProfiles(List<String> molecularProfileIds, List<String> sampleIds, List<Integer> entrezGeneIds, List<Integer> alterations) {
+
+        List<MolecularProfile> molecularProfiles = molecularProfileRepository.getMolecularProfiles(molecularProfileIds, PersistenceConstants.ID_PROJECTION);
+        Set<String> studyIds = molecularProfiles.stream().map(MolecularProfile::getCancerStudyIdentifier).collect(Collectors.toSet());
+            
+        Dataset<Row> cnaData = parquetLoader.loadStudyFiles(spark, new HashSet<>(studyIds), "/" + ParquetConstants.DATA_CNA, false);
+        if (!CollectionUtils.isEmpty(alterations)) {
+            cnaData = cnaData.where(cnaData.col("VALUE").isin(alterations.toArray()));
+        }
+        // filter based on moleclarProfileIds
+        Dataset<Row> casels = parquetLoader.loadCaseListFiles(spark, new HashSet<>(molecularProfileIds), false);
```

Comment:
what's the data format for Case List Files here? I think there is no `molecularProfileId in the case list file https://github.com/cBioPortal/cbioportal/blob/0a9923eaf5679ce40544dbace5fb397b5b133d00/persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/util/ParquetLoader.java#L62

Reply:
so for case lists, i'm taking the stableID as the filename, e.g. msk_impact_2017_cna. Does that make sense or is it incorrect?

Reply:
stableID of case list or molecular profile? sometime both might not match. example for mutations, molecular profile stable would be `studyId_mutations` and case list stable id would be `studyId_sequenced`.

Is there any change in file formats from the standard cbio ones?

Reply:
And could you point me to scripts where data files are converted into parquet files

Reply:
[ParquetWriter](https://github.com/doori/cbioportal/blob/cna-genes/persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/util/ParquetWriter.java) is the script. It is possible to write case list stable id `studyId_sequenced` to `studyId_mutations` parquet.. let me know if there's a better way to handle this.